org.eclipse.xtext/src/org/eclipse/xtext/build/IncrementalBuilder.java

```diff
@@ -311,12 +311,20 @@ protected boolean validate(Resource resource) {
 		 */
 		protected void generate(Resource resource, BuildRequest request, Source2GeneratedMapping newMappings) {
 			IResourceServiceProvider serviceProvider = getResourceServiceProvider(resource);
+			URIBasedFileSystemAccess fileSystemAccess = createFileSystemAccess(serviceProvider, resource);
+			fileSystemAccess.setContext(resource);
+			if (request.isWriteStorageResources() && resource instanceof StorageAwareResource) {
+				IResourceStorageFacade resourceStorageFacade = ((StorageAwareResource) resource)
+						.getResourceStorageFacade();
+				if (resourceStorageFacade != null) {
+					resourceStorageFacade.saveResource((StorageAwareResource) resource, fileSystemAccess);
+				}
+			}
 			GeneratorDelegate generator = serviceProvider.get(GeneratorDelegate.class);
 			if (generator == null) {
```

Comment:
wonder how this can happen

Reply:
looks like https://github.com/eclipse/xtext/commit/d1d683218487a85a968ec472a13fe40dc65f5af1
before it was IGenerator2, now its GeneratorDelegate, first can be null, but latter?

Reply:
=> simply remove null check?